pysmt/smtlib/parser/parser.py

```diff
@@ -1128,6 +1150,42 @@ def _cmd_assert(self, current, tokens):
         self.consume_closing(tokens, current)
         return SmtLibCommand(current, [expr])
 
+    def _cmd_assert_soft(self, current, tokens):
+        """(assert-soft <term> [:id <string>] [:weight <const_term>])"""
+        param_presence = {":weight": True, ":id": True}
+        expr = self.get_expression(tokens)
+        mgr = self.env.formula_manager
+        weight_value = mgr.Int(1)
+        group_clause_id = "I"  # default identifier for soft-clause
+        curr = tokens.consume()
+        params = []
+        while curr != ")":
+            tokens.add_extra_token(curr)
+            curr_parse = self.parse_atom(tokens, current)
+            if curr_parse == ":weight" and param_presence[curr_parse]:
+                weight_value = self.get_expression(tokens)
+                params.append((curr_parse, weight_value))
+                param_presence[curr_parse] = False
+            elif curr_parse == ":id" and param_presence[curr_parse]:
+                group_clause_id = self.parse_atom(tokens, "assert-soft")
+                params.append((curr_parse, group_clause_id))
+                param_presence[curr_parse] = False
+            else:
+                raise NotImplementedError("The current option is not currently implemented %s "
+                                          "or some of them are repeated" % curr_parse)
+            curr = tokens.consume()
+        if param_presence[":weight"]:
+            params.append((":weight", weight_value))
+        if param_presence[":id"]:
+            params.append((":id", group_clause_id))
+        return SmtLibCommand(current, [expr, params])
+
+    def _cmd_check_allsat(self, current, tokens):
```

Comment:
This is not part of http://optimathsat.disi.unitn.it/pages/smt2reference.html . If possible, let's update the OMT reference page, so we have a single place where to "point" to explain the extension.

Reply:
That command is inherited from mathsat 
https://mathsat.fbk.eu/smt2examples.html

Reply:
I would be careful with the extensions to the parser that are not part of the SMTLIB format. We need to keep track of where they are coming from, so that we know how to deal with potential conflicts in the future.

I can prepare a patch for the documentation to include both this change and the `to_bv` change above (in addition to the OMT stuff).